lib/vcert.rb

```diff
@@ -61,31 +73,73 @@ def renew(*args)
     # @param [Integer] timeout
     # @return [Certificate]
     def request_and_retrieve(req, zone, timeout: TIMEOUT)
+      LOG.info("#{VCERT_PREFIX} Requesting and retrieving Certificate: [#{req}], [#{zone}]")
       request zone, req
-      cert = retrieve_loop(req, timeout: timeout)
-      return cert
+      retrieve_loop(req, timeout: timeout)
     end
 
     def retrieve_loop(req, timeout: TIMEOUT)
-      t = Time.new() + timeout
+      t = Time.new + timeout
       loop do
-        if Time.new() > t
-          LOG.info("Waiting certificate #{req.id}")
+        if Time.new > t
+          LOG.info("#{VCERT_PREFIX} Waiting certificate #{req.id}")
           break
         end
         certificate = @conn.retrieve(req)
-        if certificate != nil
-          return certificate
-        end
+        return certificate unless certificate.nil?
         sleep 10
       end
-      return nil
+      nil
+    end
+
+  end
+
+  # This class provides an easy way to configure and retrieve a connector for a Venafi platform.
+  # It supports the use of token authentication for TPP, and drops the use of user/password credentials.
+  # Usage:
+  # TPP:
+  #     VenafiConnection.new url: TPP_TOKEN_URL, user: TPPUSER, password: TPPPASSWORD, trust_bundle: TRUST_BUNDLE
+  #     VenafiConnection.new url: TPP_TOKEN_URL, access_token: TPP_ACCESS_TOKEN, trust_bundle: TRUST_BUNDLE
+  #     VenafiConnection.new url: TPP_TOKEN_URL, refresh_token: TPP_REFRESH_TOKEN, trust_bundle: TRUST_BUNDLE
+  # CLoud:
+  #     VenafiConnection.new token: CLOUD_API_KEY
+  #
+  class VenafiConnection < Connection
+
+    def initialize(url: nil, access_token: nil, refresh_token: nil, user: nil, password: nil, cloud_token: nil, trust_bundle:nil, fake: false)
```

Comment:
I think this is our opportunity to fix the inappropriately named Cloud "token".  So instead of `cloud_token` in this function let's correct it to `apikey`.

Reply:
done, also changed the variable name in the cloudConnection class